README.md

```diff
@@ -1,80 +1,229 @@
+      auto_delete_resources: 'true' to delete resource when branch is deleted
+```
+
+### Github Configuration
+
+* **`api_username`**
+* **`api_token`**
+* **`push_source_to`**
+* **`branch`**
+* **`tag`**
+* **`webhook_secret`**
+* **`diff_point`**
+
+### Transifex Configuration
+
+* **`tx_config`**
+* **`api_username`**
+* **`api_password`**
+* **`push_translations_to`**
+* **`protected_branches`**
+* **`webhook_secret`**
+* **`auto_delete_resources`**
+
+### Loading Config
+
+Txgh supports two different ways of accessing configuration, raw text and a file path. In both cases, config is passed via the `TXGH_CONFIG` environment variable. You'll prefix the raw text or file path with the appropriate scheme, `raw://` or `file://`, to indicate which strategy Txgh should use.
+
+#### Raw Config
+
+Passing raw config to Txgh can be done like this:
+
+```bash
+export TXGH_CONFIG="raw://big_yaml_string_here"
 ```
-+- app
-    |
-    +- app.rb # the main Sinatra app, includes both web service endpoints
-|
-+- config
-    |
-    +- key_manager.rb # library responsible for loading config
-    |
-    +- tx.config # this is the Transifex flavored resource map file
-    |
-    +- txgh.yml # this is the main application configuration
-|
-+- lib  # all the libraries
-    |
-    + ...
-+- tests # rack/test integration tests
-    |
-    + ...
-|
-+- Rakefile # bootstrap for TravisCI
-|
-+- bootstrap.rb # includes for application paths
-|
-+- config.ru bootstrap for Sinatra
+
+When Txgh runs, it will parse the YAML payload that starts after `raw://`.
+
+#### File Config
+
+It might make more sense to store all your config in a file. Pass the path to Txgh like this:
+
+```bash
+export TXGH_CONFIG="file://path/to/config.yml"
 ```
 
+When Txgh runs, it will read and parse the file at the path that comes after `file://`.
 
-How it works
+Of course, in both the file and the raw case, environment variables can be specified via `export` or inline when starting Txgh. See the "Running Txgh" section below for more information.
+
+Running Txgh
 ---
 
-You configure a service hook in Github and point it to this server. The URL path to the service hook endpoint: /hooks/github
-You do the same for Transifex, in your project settings page, and point it to the service hook endpoint: /hooks/transifex
+Txgh is distributed as a [Docker image](https://quay.io/repository/lumoslabs/txgh) and as a [Rubygem](https://rubygems.org/gems/txgh). You can choose to run it via Docker, install and run it as a Rubygem, or run it straight from a local clone of this repository.
 
-Currently there are 4 use cases that are supported:
+### With Docker
 
-1) When a resource (configured in this service) in Transifex reaches 100% translated, the Txgh service will pull the translations and commit them to the target repository.
+Using Docker to run Txgh is pretty straightforward (keep in mind you'll need to have the Docker client set up wherever you want to run Txgh).
```

Comment:
docker server. the client is the CLI or API that tells the server to do stuff like run or pull